flocker/provision/_ssh/_model.py

```diff
@@ -52,15 +52,20 @@ class Run(PRecord):
     :ivar bytes command: The command to run.
     :ivar callable log_command_filter: A filter to apply to any logging
         of the executed command.
+    :ivar bytes pre_command: A command to run before the ``command``,
+        generally to setup the environment for the command.
```

Comment:
Each command runs in a separate channel, so the results of a `source` command are lost to subsequent commands.  This solution re-creates the environment for each subsequent command, but we don't want `source...` to appear multiple times in the docs.

The alternative solution is probably to change the RunRemotely implementation for conch to use a single CommandProtocol for multiple commands, so that the one ssh channel is used and the environment is kept between related commands.

I will consider the approach of setting a virtualenv instead of a generic pre_command. That sounds OK. The only problem is that the virtualenv locaiton is actually a directory rather than a global label. If we were using `mkvirtualenv` it would be more of a global label.